### PR TITLE
Split Discord bridge in multiple files

### PR DESCRIPTION
**Pull request by Helcaraxan**
Mainly just a copy-and-paste. Only real code changes:
- split between channels and members RW mutex.
- better clean-up and maintenance of members map on update
- improved username mention replacement logic with accompanying unit-test

### CODE REVIEW

**Comment by Helcaraxan** [comment]
The complexity issue from CodeClimate will obviously solved in subsequent PRs to limit the scope of changes in this one.

File: bridge/discord/handlers.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	b.Log.Debugf("<= Message is %#v", rmsg)
+	b.Remote <- rmsg
+}
+
+func (b *Bdiscord) memberUpdate(s *discordgo.Session, m *discordgo.GuildMemberUpdate) {
+	if m.Member == nil {
+		b.Log.Warnf("Received member update with no member information: %#v", m)
+	}
+
+	b.membersMutex.Lock()
```
Comment (42wim) [inline bridge/discord/handlers.go]: I'm wondering if we don't have a possible locking issue/race here, we're also locking in `Connect()` where we add the `memberUpdate` handle (I know this issue exists in the original code before your PR too :-)

**Comment by 42wim** [comment]
Small question, otherwise LGTM

File: bridge/discord/handlers.go (hunk 1 above)
Comment (Helcaraxan) [reply bridge/discord/handlers.go]: No. That would not be an issue. We would need two locks (which we do have - `channelsMutex`) for which a single codepath tries to acquire one while holding the other. That could then trigger a deadlock.

In this case the code here simply runs and there is nothing inside the locked range on which the execution can block. The handler is only called if the relevant event is received by the endpoint... however even if that is the case then the handler (which will run on its own goroutine anyway triggered by the dispatcher) will simply wait on the main routine to exit the locked range and then acquire the lock for itself.

**Pull request merged**